src/main/java/hudson/remoting/Engine.java

```diff
@@ -758,11 +759,18 @@ private void innerRun(IOHub hub, SSLContext context, ExecutorService service) {
                 final JnlpAgentEndpoint endpoint;
                 try {
                     endpoint = resolver.resolve();
-                } catch (Exception e) {
-                    if (Boolean.getBoolean(Engine.class.getName() + ".nonFatalJnlpAgentEndpointResolutionExceptions")) {
-                        events.status("Could not resolve JNLP agent endpoint", e);
+                } catch (IOException e) {
```

Comment:
Minor cleanup while I was here. The original code was catching any exception type, including `InterruptedException` (which it made no attempt to handle). Following generic advice such as that given on [this page](https://stackify.com/best-practices-exceptions-java/), prefer specific exceptions in catch blocks.